net/prosody/Makefile

```
@@ -56,18 +53,13 @@ define Build/Configure
 		--with-lua="$(STAGING_DIR_HOSTPKG)" \
 		--with-lua-include="$(STAGING_DIR)/usr/include" \
 		--with-lua-lib="$(STAGING_DIR)/usr/lib" \
-		--cflags="$(TARGET_CFLAGS)" \
+		--cflags="$(TARGET_CFLAGS) $(FPIC) -std=gnu99" \
 		--ldflags="$(TARGET_LDFLAGS) -llua -lm -ldl -shared" \
 		--c-compiler="$(CC)" \
 		--linker="$(LD)" \
 		--datadir="/etc/prosody/data" \
 	)
 endef
-#	LDFLAGS="$(TARGET_LDFLAGS) -llua -lm -ldl" \
-
-MAKE_FLAGS += \
-	CFLAGS="$(TARGET_CFLAGS) $(TARGET_CPPFLAGS) -std=gnu99" \
-	PREFIX="/usr" \
 
```

```suggestion
MAKE_FLAGS += LD="$(TARGET_CC)"
```
This alone should make it work.  Cleaning up the Makefile won't hurt.  Notice that setting `--linker` above won't do it because the assignment gets overriden by `LD` already being set in `MAKE_FLAGS`.